wannierberri/calculators/static.py

```diff
@@ -25,54 +64,552 @@ def __init__(self, fder,**kwargs):
+        With Fermi surface integral.
+        Output:
+        :math: `\sigma_{\alpha\beta} = -e^2/\hbar \tau \int [dk] v_\alpha v_\beta f'`
+        Instruction:
+        :math: `j_\alpha = \sigma_{\alpha\beta} E_\beta`"""
+        super().__init__(**kwargs)
+
+
+# E^1 B^1
+class Hall_classic_FermiSurf(StaticCalculator):
+    def __init__(self, **kwargs):
+        self.Formula = frml.VelMassVel
+        self.factor = factor_t2_1_1
+        self.fder = 1
+        self.comment  = r"""Classic Hall conductivity (S/m/T)
+        With Fermi surface integral.
+        Output:
+        :math: `\sigma_{\alpha\beta :\mu} = e^3/\hbar^2 \tau^2 \epsilon_{\gamma\mu\rho} \int [dk] v_\alpha \partial_\rho v_\beta v_\gamma f'`
+        Instruction:
+        :math: `j_\alpha = \sigma_{\alpha\beta :\mu} E_\beta B_\mu`"""
+        super().__init__(**kwargs)
+
+    def __call__(self, data_K):
+        res = super().__call__(data_K)
+        res.data = res.data[:, :, :, beta_A, alpha_A] - res.data[:, :, :, alpha_A, beta_A]
+        res.data = 0.5 * (res.data[:, alpha_A, beta_A, :] - res.data[:, beta_A, alpha_A, :])
+        res.rank -= 2
+        return res
+
+
+class Hall_classic_FermiSea(StaticCalculator):
+    def __init__(self, **kwargs):
+        self.Formula = frml.MassMass
+        self.factor = factor_t2_1_1
         self.fder = 0
+        self.comment  = r"""Classic Hall conductivity (S/m/T)
+        With Fermi sea integral.
+        Output:
+        :math: `\sigma_{\alpha\beta :\mu} = -e^3/\hbar^2 \tau^2 \epsilon_{\gamma\mu\rho} \int [dk] \partial_\gamma v_\alpha \partial_\rho v_\beta f`
+        Instruction:
+        :math: `j_\alpha = \sigma_{\alpha\beta :\mu} E_\beta B_\mu`"""
         super().__init__(**kwargs)
 
+    def __call__(self, data_K):
+        res = super().__call__(data_K)
+        res.data = res.data.transpose(0, 4, 1, 2, 3)
+        res.data = res.data[:, :, :, beta_A, alpha_A] - res.data[:, :, :, alpha_A, beta_A]
+        res.data = 0.5 * (res.data[:, alpha_A, beta_A, :] - res.data[:, beta_A, alpha_A, :])
+        res.rank -= 2
+        return res
+
 
-class BerryDipole_FermiSurf(StaticCalculator):
+# E^2 B^0
+class NLAHC_FermiSurf(StaticCalculator):
 
     def __init__(self, **kwargs):
         self.Formula = frml.VelOmega
-        self.factor = 1
+        self.factor = factor_t1_2_0
         self.fder = 1
+        self.comment = r"""Nonlinear anomalous Hall conductivity  (S^2/A) with use_factor=True
+Berry curvature dipole (dimensionless) with use_factor=False
+        With Fermi surface integral. Eq(8) in `Ref <https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.115.216806>`_
+        Output:
+        :math: `D_{\beta\delta} = -e^3/\hbar^2 \tau \int [dk] v_\beta \Omega_\delta f'`
+        Instruction:
+        :math: `j_\alpha = \epsilon_{\alpha\delta\gamma} \D_{\beta\delta} E_\beta E\gamma`"""
         super().__init__(**kwargs)
 
 
-class BerryDipole_FermiSea(StaticCalculator):
+class NLAHC_FermiSea(StaticCalculator):
```

Comment:
I would still leave the calculator's name "BerryDipole_**" , it is more sound and it is nice that it is dimensionless

You may add 
```
class NLAHC_Fermisea(BerryDipole_FermiSea):
    def __init__(self,**kwargs):
        super().__init__(kwargs)
        self.factor = 
        self.comment = 
```
So, they differ only by factor and comment

Same for fermisurface.